yt_dlp/extractor/wordpress.py

```diff
@@ -67,3 +72,87 @@ def _extract_from_webpage(self, url, webpage):
+        'info_dict': {
+            'id': 'over-the-horizon-the-evolution-of-the-samsung-galaxy-brand-sound',
+            'title': 'Over the Horizon: The Evolution of the Samsung Galaxy Brand Sound',
+            'age_limit': 0,
+            'thumbnail': 'https://img.global.news.samsung.com/global/wp-content/uploads/2015/04/OTH_Main_Title-e1429612467870.jpg',
+            'description': 'md5:bc3dd738d1f11d9232e94e6629983bf7',
+        },
+        'playlist': [{
+            'info_dict': {
+                'id': 'over_the_horizon_2013',
+                'ext': 'mp3',
+                'title': 'Over the Horizon 2013',
+                'url': 'http://news.samsung.com/global/wp-content/uploads/ringtones/over_the_horizon_2013.mp3'
+            }
+        }],
+        'playlist_count': 6,
+        'params': {'skip_download': True}
+    }, {
+        # Version 1.9.3: https://plugins.trac.wordpress.org/browser/wp-miniaudioplayer/tags/1.9.3
+        'url': 'https://www.booksontape.com/collections/audiobooks-with-teacher-guides/',
+        'info_dict': {
+            'id': 'audiobooks-with-teacher-guides',
+            'title': 'Audiobooks with Teacher Guides | Books on Tape',
+            'age_limit': 0,
+            'thumbnail': 'https://www.booksontape.com/wp-content/uploads/2016/09/bot-logo-1200x630.jpg',
+        },
+        'playlist_mincount': 12
+    }, {
+        # Version 1.9.7: https://plugins.trac.wordpress.org/browser/wp-miniaudioplayer/tags/1.9.7
+        # But has spaces around href filter
+        'url': 'https://www.estudiords.com.br/temas/',
+        'info_dict': {
+            'id': 'temas',
+            'title': 'Temas Variados',
+            'age_limit': 0,
+            'timestamp': float,
+            'upload_date': str,
+            'thumbnail': 'https://www.estudiords.com.br/wp-content/uploads/2021/03/LOGO-TEMAS.png',
+            'description': 'md5:ab24d6a7ed0312ad2d466e721679f5a0',
+        },
+        'playlist_mincount': 30
+    }]
+
+    def _extract_from_webpage(self, url, webpage):
+        # Common function for the WordPress plugin version only.
+        mb_player_params = self._search_regex(
+            r'function\s*initializeMiniAudioPlayer\(\){[^}]+jQuery([^;]+)\.mb_miniPlayer',
+            webpage, 'mb player params', default=None)
+        if not mb_player_params:
+            return
+        # v1.55 - 1.9.3 has "a[href*='.mp3'] ,a[href*='.m4a']"
+        # v1.9.4+ has "a[href*='.mp3']" only
+        file_exts = re.findall(r'a\[href\s*\*=\s*\'\.([a-zA-Z\d]+)\'', mb_player_params)
+        if not file_exts:
+            return
+
+        # XXX: this is inefficient, but it works
+        candidates = (
+            get_element_text_and_html_by_tag('a', html)
+            for _, html in get_elements_text_and_html_by_attribute(
+                'href', rf'(?:[^\"\']+\.(?:{"|".join(file_exts)}))', webpage, escape_value=False))
```

Comment:
We could add an optional `tag` parameter to `get_elements_text_and_html_by_attribute`

Reply:
hmm. Could do something like

```diff
diff --git a/yt_dlp/utils.py b/yt_dlp/utils.py
index 3e2ce8434..746ac0295 100644
--- a/yt_dlp/utils.py
+++ b/yt_dlp/utils.py
@@ -408,7 +408,7 @@ def get_elements_html_by_attribute(*args, **kwargs):
     return [whole for _, whole in get_elements_text_and_html_by_attribute(*args, **kwargs)]
 
 
-def get_elements_text_and_html_by_attribute(attribute, value, html, escape_value=True):
+def get_elements_text_and_html_by_attribute(attribute, value, html, tag=r'[a-zA-Z0-9:._-]+', escape_value=True):
     """
     Return the text (content) and the html (whole) of the tag with the specified
     attribute in the passed HTML document
@@ -419,7 +419,7 @@ def get_elements_text_and_html_by_attribute(attribute, value, html, escape_value
     value = re.escape(value) if escape_value else value
 
     partial_element_re = rf'''(?x)
-        <(?P<tag>[a-zA-Z0-9:._-]+)
+        <(?P<tag>{tag})
          (?:\s(?:[^>"']|"[^"]*"|'[^']*')*)?
          \s{re.escape(attribute)}\s*=\s*(?P<_q>['"]{quote})(?-x:{value})(?P=_q)
         '''

```

Reply:
looks fine